documentation/Amazon-Managed-Grafana/ReadMe.md

```diff
@@ -0,0 +1,149 @@
+apiVersion: external-secrets.io/v1beta1
+kind: ClusterSecretStore
+metadata:
+  name: grafana-secret-store
+spec:
+  provider:
+    aws:  # set secretStore provider to AWS.
+      service: SecretsManager # Configure service to be Secrets Manager
+      region: us-west-2   # Region where the secret is.
+      auth:
+        secretRef:
+          accessKeyIDSecretRef: 
+            name: aws-secret # References the secret we created
+            namespace: default
+            key: access-key  
+          secretAccessKeySecretRef:
+            name: aws-secret
+            namespace: default
+            key: secret
+EOF
+```
+
+> Finally, Grafana operator should be enhanced to read secrets like below :
+
+```.bash
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: grafana-token-secret
+  namespace: grafana
+spec:
+  refreshInterval: 1m
+  secretStoreRef:
+    name: grafana-secret-store #The secret store name we have just created.
+    kind: ClusterSecretStore
+  target:
+    name: grafana-secret # Secret name in k8s
+  data:
+  - secretKey: grafana-token # which key it's going to be stored
+    remoteRef:
+      key: grafana-api-key # Our secret-name goes here
+```
+
+### Adding AWS data sources to Amazon Managed Grafana
+
+Today Grafana operator supports [Amazon Managed Service for Prometheus](https://github.com/grafana-operator/grafana-operator/blob/master/deploy/examples/datasources/AWS-Prometheus.yaml) as a data source. Grafana operator solution should be enhanced to add following AWS data sources (minimally Amazon CloudWatch) :
+- Amazon CloudWatch
+- Amazon OpenSearch Service
+- AWS IoT SiteWise
+- AWS IoT TwinMaker
+- Amazon Timestream
+- Amazon Athena
+- Amazon Redshift
+- AWS X-Ray
+
+This is not a road block feature but nice to have to support all AWS data sources.
+
+In order to accomplish this, "Reading keys for accessing Amazon Managed Grafana from AWS Secrets Manager" feature above becomes a prerequisite. `GrafanaDataSource` CRD should be modified to use `grafana-secret` to add AWS data sources as `GrafanaDataSource` on Amazon Managed Grafana. This allows customers to use GitOps approach to add AWS data sources as `GrafanaDataSource` on remote Grafana instances such as Amazon Managed Grafana.
+
+### Creating Grafana dashboards on Amazon Managed Grafana.
+
+Today Grafana operator supports the creation of Grafana dashboards on self managed grafana instance in the cluster where the Grafana operator is installed. Grafana operator should be enhanced to support creating dashboards on Amazon Managed Grafana. 
+
+In order to accomplish this, "Reading keys for accessing Amazon Managed Grafana from AWS Secrets Manager" feature above becomes a prerequisite. `GrafanaDashboard` CRD should be modified to use `grafana-secret` to create `GrafanaDashboard` on Amazon Managed Grafana. This allows customers to use GitOps approach to create Grafana Dashboards on remote Grafana instances such as Amazon Managed Grafana.
```

Comment:
> So I think we misunderstood each other, to me this is a long list of issues instead of a design proposal.
> 
> We need to have a basic discussion on how to design the CRD:s to support external grafana instances. Assuming that the secrets needed to connect to your external grafana instance is available inside the cluster how would we use these secrets to talk to the external grafana instance?
> 
> How would the grafana crd look like? How would the datasource crd look like?
> 
> If you don't feel comfortable with go that is fine, we will figure out the controller part but as an end user how would you like the operator work with external resources?

Thankyou for responding @NissesSenap We can definately jump in to a call to discuss on open items and next steps and we can also discuss over slack. From and enduser perspective, the operator should be able to use the secrets in the cluster and access external grafana instance to add data sources like AMP, CW and also create Dashboards and alerting on remote AMG using same or new CRDs.